Add composer autoload

MaxwellGarceau opened the pull request:
<!--
Filling out this template is required.  Any PR that does not include enough information to be reviewed may be closed at a maintainers' discretion.  All new code requires documentation and tests to ensure against regressions.
-->

### Description of the Change
<!--
We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much detail as possible.

Where possible, please also include:
- verification steps to ensure your change has the desired effects and has not introduced any regressions
- any benefits that will be realized
- any alternative implementations or possible drawbacks that you considered
- screenshots or screencasts
-->

Added autoloading to have an easy way to modularize new code and refactor existing code in the future.

**❗I modified the Github action workflow files to install composer dependencies in order to ensure autoload was being generated on every build. I wanted to flag this in case I missed a required build process update.**

Also, if there's a good place to add documentation for in progress pattern updates I'd be happy to add switching to autoload in new work and any refactors.

<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
Closes #96 

### How to test the Change
<!-- Please provide steps on how to test or validate that the change in this PR works as described. -->

### Changelog Entry
<!--
Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR. -->
> Developer - Add Composer autoload.

### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @username, @username2, ...

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/mailchimp/wordpress/blob/develop/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

dkotter commented on mailchimp.php:
```diff
@@ -34,6 +34,17 @@
+
+// Check if the autoload file exists
+if ( ! file_exists( $autoload_file ) ) {
+	die( 'Composer autoload file not found. Run `composer install` to generate it.' );
```
Instead of using `die` here, I'd suggest instead we use a normal WordPress admin notice for a better user experience. Can see an example of that here: https://github.com/10up/restricted-site-access/blob/develop/restricted_site_access.php#L32-L58

MaxwellGarceau replied:
Good idea and thanks for the link! I just added the admin notice.

dkotter reviewed (changes requested):
Overall this looks good but we'll need to do a few more things:

1. We have a GitHub Action that deploys plugin assets and readme changes. That will need to be updated to run `composer install` as well
2. We don't currently deploy the `vendor` directory so we'll need to update that or things won't work in the final release. We have both a `.distignore` and `.gitattributes` files that will need updated

MaxwellGarceau commented:
> Overall this looks good but we'll need to do a few more things:
> 
> 1. We have a GitHub Action that deploys plugin assets and readme changes. That will need to be updated to run `composer install` as well
> 2. We don't currently deploy the `vendor` directory so we'll need to update that or things won't work in the final release. We have both a `.distignore` and `.gitattributes` files that will need updated

@dkotter I've made the requested changes.
1. Updated [.github/workflows/wordpress-plugin-asset-update.yml](https://github.com/mailchimp/wordpress/pull/95/files#diff-c3e586c1c6d72ed01af94b58e01c947574f715cca5273395f6d5710d604bcf49) with the `composer install` and autodump commands
2. Removed the vendor folder from distignore and gitattributes

qasumitbagthariya commented:
### QA Update  ✅
---

- Run composer install to install dependencies and generate the autoload files. (Composer version 2.8.4) ✅

- Verify that the vendor/autoload.php file is generated. ✅

- Verify Functionality of Existing Features


<img width="598" alt="image" src="https://github.com/user-attachments/assets/0ac52aea-f28d-46bc-bbda-8a94b137f4a4" />

qasumitbagthariya commented:
Regression / Smoke Test Report  ✅ 
-------

Tested with the `smoke-testing` branch, it works as expected, similar to the fix-specific branch.

**Testing Environment** 

<Details>

- WordPress: 6.7.1
- Theme: Storefront 4.6.0
- Theme: Twenty Twenty-Four 1.3
- WooCommerce - 9.5.1
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 
- Branch: smoke-testing

</Details>

vikrampm1 commented:
@dkotter got it. @iamdharmesh can you able to assist with the above as @MaxwellGarceau is not resourced on this project anymore?

iamdharmesh commented:
@dkotter E2E tests were failing because we are using the build release action from the develop branch, and it doesn't have the composer installation step. Due to an early return in the autoload file check, the plugin functionality was not getting enabled. I temporarily [pointed](https://github.com/mailchimp/wordpress/pull/95/commits/82116e0e55e2f429b2b6813d897fe0f3a6ef2eae) the build release action to this PR, and it started working as expected.

However, we recently enabled 2FA on the OSP account, and because of this, the connection tests and tests requiring connections are failing. I tried running the tests locally with a different account, and they worked well. So, it seems we are good to proceed here.

cc: @vikrampm1
